mbs_results/outputs/selective_editing_question_output.py

```diff
@@ -0,0 +1,153 @@
+        dataframe=df_with_domain,
+        reference=reference,
+        period=period,
+        domain=domain,
+        question_no=question_no,
+        predicted_value=predicted_value,
+        imputation_marker="imputation_flags_adjusted_value",
+        a_weight=a_weight,
+        o_weight=o_weight,
+        g_weight=g_weight,
+        auxiliary_value=aux,
+        period_selected=period_selected,
+    )
+
+    standardising_factor["survey_code"] = "009"
+
+    # TODO Update file name
+    standardising_factor["imputation_flags_adjusted_value"] = standardising_factor[
+        "imputation_flags_adjusted_value"
+    ].str.upper()
+    standardising_factor = standardising_factor.rename(
+        columns={
+            "reference": "ruref",
+            "domain": "domain_group",
+            aux: "auxiliary_value",
+            "imputation_flags_adjusted_value": "imputation_marker",
+            question_no: "question_code",
+        }
+    )
+
+    return standardising_factor
+
+
+def validation_checks_selective_editing(df: pd.DataFrame):
+    """
+    validation checks for duplicate rows and number of NaNs
+    Currently prints to console.
+
+    Parameters
+    ----------
+    df : pd.DataFrame
+        dataframe output from `create_selective_editing_question_output`
+    """
+    print(
+        "Number of duplicates, (checking period, question_no, and reference:",
+        df.duplicated(subset=["period", "question_code", "ruref"]).sum(),
+    )
+    predicted_na = df.loc[df["predicted_value"].isna()]
+    number_nans = predicted_na.count()[0]
+    print(f"predicted_values has {number_nans} NaNs")
+
+
+if __name__ == "__main__":
+    from datetime import datetime
+
+    version = metadata.metadata("monthly-business-survey-results")["version"]
+    output_path = (
+        "C:/Users/dayj1/Office for National Statistics/Legacy Uplift - MBS (1)/"
+    )
+    wins_output = pd.read_csv(
+        output_path + f"winsorisation/winsorisation_output_{version}.csv"
+    )
+    previous_period = 202201
+    output = create_selective_editing_question_output(
+        df=wins_output,
+        reference="reference",
+        period="period",
+        domain="domain",
+        question_no="question_no",
+        sic="sic_5_digit",
+        aux="frotover",
+        a_weight="design_weight",
+        o_weight="outlier_weight",
+        g_weight="calibration_factor",
+        predicted_value="predicted_value",
+        imputed_value="imputed_value",
+        adjusted_value="adjusted_value",
+        output_path=output_path,
+        period_selected=previous_period,
+    )
+    validation_checks_selective_editing(output)
+    formatted_date = datetime.today().strftime("%Y-%m-%d")
+    output_file_name = (
+        f"sopp_mbs_{formatted_date}_selective_editing"
+        + f"_contributor_{previous_period}_{version}.csv"
+    )
+    output.to_csv(
+        output_path + "selective_editing_outputs/" + output_file_name,
+        index=False,
+    )
```

Comment:
This is how the user will call the outputs function right ?  We could remove the main bit and add it as an example how to use it ?